packages/salesforcedx-vscode-soql/src/queryDataView/dataProviders/csvDataProvider.ts

```diff
@@ -26,4 +27,75 @@ export class CsvDataProvider implements DataProvider {
   public getFileName(): string {
     return `${this.documentName}.${this.fileExtension}`;
   }
+
+  protected toTable(query: string, data: JsonMap[]): Papa.UnparseObject {
+    const columnData = new SelectAnalyzer(query).getColumnData();
+
+    const fields: string[] = [];
+    const colDataQ = [columnData];
+    while (colDataQ.length > 0) {
+      const cd = colDataQ.shift();
```

Comment:
Array `shift` is an O(n) complexity operation. Consider using a `Set` data structure for this